Package.swift

```diff
@@ -4,6 +4,16 @@
 
 import PackageDescription
 
+#if os(WASI)
```

Comment:
Does this actually have the desired effect? `Package.swift` is evaluated on the host platform, not on the target platform, so `#if os(WASI)` condition will never be evaluated until we have a self-hosted toolchain running in the browser (we probably won't for long time). These kind of exclusion of dependencies should be implemented with `condition: .when(platforms: [.wasi])`, as done for the `TokamakShim` target below.

This check also seems to be the reason for the Linux build failing.

Reply:
Oops! 😂 

Although conditions don't seem to work, as we can't depend on a version # of JavaScriptKit since it has unsafe build flags, unless I'm doing something wrong? Should I just change JavaScriptKit to a commit?

Reply:
Are you using a recent 5.3 SwiftWasm snapshot as recorded in the `.swift-version` file in `main`? These snapshots allow adding JSKit with a version as a dependency, even with unsafe flags.

Reply:
I see what you mean, Swift toolchains for other platforms don't have the unsafe flags workaround. I hope that if you don't attempt to build products that depend on JavaScriptKit, it won't complain about unsafe flags? As in `swift build --product TokamakStaticHTML`, or when you build something that depends on Tokamak, but doesn't depend on `TokamakDOM`?

Reply:
In my publish test project it seems to resolve the dependencies for all the products, not just TokamakStaticHTML...

Reply:
That's probably because https://github.com/apple/swift-package-manager/pull/2749 was merged after Swift 5.3 was branched off. Guess we're stuck with this behavior until the next version of Swift. 😞

IDK what's the best workaround here, maybe we could get rid of unsafe flags in JavaScriptKit somehow?

@kateinoigakukun what do you think? The gist is that packages that depend on JavaScriptKit can't be used on non-Wasm platforms due to the use of unsafe flags. That's because Swift 5.3 resolves all dependencies regardless of whether they're used or not. It even ignores platform conditions apparently. Do you think we could find a way to build JavaScriptKit without unsafe flags by modifying our SwiftWasm 5.3 branch? Maybe we could make it infer linker flags automatically?

Reply:
I think we need to add new `LinkerSetting` cases in SwiftPM to support exporting and importing symbols.

Reply:
Even if we add new `LinkerSetting` values, we still need upstream to support them. Now is the best time to submit a PR with these settings upstream before the new major version is branched off, but it means that we still can't create a package manifest that declares these new safe linker settings consumable with upstream Swift 5.3. Or am I missing something?

Reply:
The only way we can take to avoid this issue is to separate this repository into a package that depends on JavaScriptKit and a package that doesn't depends on JavaScriptKit.

Reply:
As far as I know, Swift 5.3 still downloads the whole tree of dependencies without checking whether they're used or not. This was fixed in apple/swift-package-manager#2749, but it was merged after Swift 5.3 was branched off.

Reply:
I can't figure out another idea 🤷 The user must use wasm toolchain now.

Reply:
A horrible hack would be to patch SwiftPM to check whether it's linking JavaScriptKit and to inject the required linker flags. This probably isn't worse than what we do now, as SwiftPM in 5.3 SwiftWasm snapshots already checks for a package name and allows unsafe flags if it's JavaScriptKit. But submitting a PR upstream to allow more linker flags is a priority then. I don't want to miss the next version cut-off again. 🙏